pkg/cmd/adm/register_member.go

```diff
@@ -120,66 +114,210 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
 		return nil
 	}
 
-	return validated.perform(ctx, newCommand)
+	return validated.perform(ctx)
+}
+
+func (v *registerMemberValidated) getSourceAndTargetClusters(sourceClusterType configuration.ClusterType) (clusterData, clusterData) {
+	if sourceClusterType == configuration.Member {
+		return v.memberClusterData, v.hostClusterData
+	}
+	return v.hostClusterData, v.memberClusterData
 }
 
-func runAddClusterScript(term ioutils.Terminal, newCommand client.CommandCreator, joiningClusterType configuration.ClusterType, hostKubeconfig, hostNs, memberKubeconfig, memberNs, nameSuffix string, useLetsEncrypt bool) error {
-	if !term.AskForConfirmation(ioutils.WithMessagef("register the %s cluster by creating a ToolchainCluster CR, a Secret and a new ServiceAccount resource?", joiningClusterType)) {
+// addCluster creates a secret and a ToolchainCluster resource on the `targetCluster`.
+// This ToolchainCluster CR stores a reference to the secret which contains the kubeconfig of the `sourceCluster`. Thus enables the `targetCluster` to interact with the `sourceCluster`.
+// - `targetCluster` is the cluster where we create the ToolchainCluster resource and the secret
+// - `sourceCluster` is the cluster referenced in the kubeconfig/ToolchainCluster of the `targetCluster`
+func (v *registerMemberValidated) addCluster(ctx *extendedCommandContext, sourceClusterType configuration.ClusterType) error {
+	if !ctx.AskForConfirmation(ioutils.WithMessagef("register the %s cluster by creating a ToolchainCluster CR, a Secret and a new ServiceAccount resource?", sourceClusterType)) {
 		return nil
 	}
+	sourceClusterDetails, targetClusterDetails := v.getSourceAndTargetClusters(sourceClusterType)
+	// wait for the SA to be ready
+	toolchainClusterSAKey := runtimeclient.ObjectKey{
+		Name:      fmt.Sprintf("toolchaincluster-%s", sourceClusterType),
+		Namespace: sourceClusterDetails.namespace,
+	}
+	if err := waitForToolchainClusterSA(ctx.CommandContext, sourceClusterDetails.client, toolchainClusterSAKey, v.args.waitForReadyTimeout); err != nil {
+		ctx.Printlnf("The %s ServiceAccount in the %s cluster is not present.", toolchainClusterSAKey, sourceClusterType)
+		ctx.Printlnf("Please check the %s ToolchainCluster ServiceAccount in the %s %s cluster.", toolchainClusterSAKey, sourceClusterDetails.apiEndpoint, sourceClusterType)
+		return err
+	}
+	// source cluster details
+	ctx.Printlnf("API endpoint retrieved: %s", sourceClusterDetails.apiEndpoint)
+	ctx.Printlnf("joining cluster name: %s", sourceClusterDetails.toolchainClusterName)
```

Comment:
```suggestion
	ctx.Printlnf("The source cluster name: %s", sourceClusterDetails.toolchainClusterName)
	ctx.Printlnf("The API endpoint of the source cluster: %s", sourceClusterDetails.apiEndpoint)
```

Reply:
I wasn't sure we wanted to change the wording of the logs too or keep them as is for now. Thanks!